packages/connection-manager/src/js/Connection/AbstractConnection.js

```diff
@@ -0,0 +1,91 @@
+import { EventEmitter } from "events";
```

Comment:
Let's please reduce the number of possible states to make it easier to reason about a connections state.